File: servers/assessment/assessments/assessmentDTO/studentAssessment.go
package assessmentDTO

import (
	"github.com/google/uuid"
	"github.com/ls1intum/prompt2/servers/assessment/assessments/assessmentCompletion/assessmentCompletionDTO"
	"github.com/ls1intum/prompt2/servers/assessment/assessments/scoreLevel/scoreLevelDTO"
	"github.com/ls1intum/prompt2/servers/assessment/evaluations/evaluationDTO"
)

type StudentAssessment struct {
	CourseParticipationID uuid.UUID                                    `json:"courseParticipationID"`
	Assessments           []Assessment                                 `json:"assessments"`
	AssessmentCompletion  assessmentCompletionDTO.AssessmentCompletion `json:"assessmentCompletion"`
	StudentScore          scoreLevelDTO.StudentScore                   `json:"studentScore"`
	SelfEvaluations       []evaluationDTO.Evaluation                   `json:"selfEvaluations"`
	PeerEvaluations       []evaluationDTO.Evaluation                   `json:"peerEvaluations"`
}

File: clients/assessment_component/src/assessment/interfaces/studentAssessment.ts
import { Assessment } from './assessment'
import { AssessmentCompletion } from './assessmentCompletion'
import { StudentScore } from './studentScore'
import { Evaluation } from './evaluation'

export type StudentAssessment = {
  courseParticipationID: string // UUID
  assessments: Assessment[]
  assessmentCompletion: AssessmentCompletion
  studentScore: StudentScore
  selfEvaluations: Evaluation[]
  peerEvaluations: Evaluation[]
}
